pkg/cluster/cluster.go

```diff
@@ -206,15 +207,33 @@ func (c *Clusterer) Start(ctx context.Context) error {
 		}
 
 		// We now have either joined or started a new cluster.
-		// Nodes initially join in the Viewer state. We can move to the
-		// Participant state to signal that we wish to participate in reading
+		// Nodes initially join in the Viewer state. We wait for the readyFunc
+		// to verify that the graph has been evaluated to move on to the
+		// Participant state and signal that we wish to participate in reading
 		// or writing data.
-		ctx, ccl := context.WithTimeout(ctx, 5*time.Second)
-		defer ccl()
-		err = node.ChangeState(ctx, peer.StateParticipant)
-		if err != nil {
-			return err
-		}
+		go func(ctx context.Context) {
```

Comment:
Are there any benefits of doing this instead of extracting the call to ChangeState to a new method which we invoke when we know the load has finished?

Reply:
You're right, I was overthinking it, KISS :)

The errors that can return are either due to invalid transitions or not being able to encode the message, both of which wouldn't be solved with a retry.